lib/templatesimplifier.h

```diff
@@ -242,6 +242,7 @@ class CPPCHECKLIB TemplateSimplifier {
         bool isSameFamily(const TemplateSimplifier::TokenAndName &decl) const {
             // Make sure a family flag is set and matches.
             // This works because at most only one flag will be set.
+            // NOLINTNEXTLINE(misc-redundant-expression)
             return ((mFlags & fFamilyMask) & (decl.mFlags & fFamilyMask)) != 0;
```

Comment:
I think it's ok to fix this warning

Reply:
It's a false positive. I still need to file a ticket about this.

Reply:
I assume it complains that it's redundant to repeat `fFamilyMask` right? So the fix would be either something like:

    return (mFlags & decl.mFlags & fFamilyMask) != 0;

or:

    return (mFlags & fFamilyMask) && (decl.mFlags & fFamilyMask);

Reply:
I did not look closely at the warning before. It is indeed about the redundant `fFamilyMask`.

I chose the send suggestion for readability.